xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm

```diff
@@ -1120,7 +1120,7 @@ $html
   #foreach($docName in $docNames)
     #if($xwiki.hasAccessLevel("view", $xcontext.user, $docName))
       #set($document = $xwiki.getDocument($docName).getTranslatedDocument())
-      #if(!$blacklistedSpaces.contains($document.getSpace()))
+      #if(!$document.isHidden() && !$blacklistedSpaces.contains($document.getSpace()))
```

Comment:
Again not a fan of this new criteria.

Reply:
I explained it in the description of the PR: 
> Note that I kept the $blacklistedSpaces parameter of the macro displayDocumentList since it could still be used but the condition now check both that parameter and the hidden property.

Reply:
```suggestion
      #if(("$!blacklistedSpaces" != '' && !$blacklistedSpaces.contains($document.getSpace())) || ("$!blacklistedSpaces" == '' && ($services.user.properties.displayHiddenDocuments() || !$document.isHidden())))
```